core/state/statedb.go

```diff
@@ -199,6 +204,19 @@ func (self *StateDB) Logs() []*types.Log {
 	return logs
 }
 
+func (self *StateDB) AddPreimage(hash common.Hash, preimage []byte) {
```

Comment:
Please document this. It should say that the method is used for recording preimages seen by the VM.